### ISSUE TITLE
High Res Optimization on Vertical aspect ratios

### ISSUE THREAD

**Issue by movinimage** [invoke-ai/InvokeAI#1236]
I have been playing with HRO in the GUI. When I choose a vertical aspect ratio for the final size, and look in the terminal I can see that the initial image (pre upscale with ddim) is being rendered with a horizontal (the short side) dimension of less than 512. From what I understand - and from my testing, having a dimension of less than 512 often results in poor, primitive output. I've found that in this situation, I get unusable results with Manny of my prompts. The output is often cartoony and like primitive art...similar to the original DALLE or other earlier models.

**Comment by psychedelicious** [comment]
I had discussed this with another user on discord - can't remember who - and we agreed that the choice of initial image should do its best not to use lower than 512px for either dimension. In my experience it's more like 384px where the issue becomes apparent.

**Comment by movinimage** [comment]
Another idea - Because the image with HRO will be different from the image with the same final output dimensions with HRO off, it is a gamble to choose HRO and then wait for the whole process to see what it delivers. HRO is slow, so going through multiple variations becomes tedious.

Thinking about workflow, and that most people like to see a number of variations before making their final choice, It might be nice to be able to do a set of initial variations at the size HRO will use for the initial image, then pick one before the HRO ddim upscale. Therefore, upscaling only once instead of with every variation when HRO is turned on.

I can do this manually by 
-Toggling HRO on with my final output res selected.
-Pressing the Invoke button
-seeing what HRO chooses for an initial size in the terminal 
-Canceling the Invoke
-turn off HRO
-Invoke a set of variations at the initial dimensions that HRO had chosen.
-choose a variation I like, then copy the seed, etc... from that variation
-then, reset the final output rez, toggle HRO on, and Invoke to get my final image.

This saves time but is a bit cumbersome.

Just thinking. :). I love the work being done here. By far the best looking UI and user experience in the SD world.

**Comment by psychedelicious** [comment]
You raise a very good point, I also wondered why I can't just HRO an existing image. I think it's because it needs the image creation parameters to do it? Unsure. Will defer to other contributors for this.

**Comment by psychedelicious** [comment]
Another issue is that the step_callback doesn't indicate which stage of the HRO process it is on, this cause a bug in the web ui with the progress bar. We need the HRO phase to be included in the step callback.

**Comment by cwallen** [comment]
I was also having issues with hires being less clear than it should be. 
https://github.com/invoke-ai/InvokeAI/pull/1299 Might help here as well.

**Comment by psychedelicious** [comment]
This is partially resolved by a recent PR - can't remember which one - the calculation for HRO dimensions was improved.

But the request for more control over the HRO process is not yet complete.

**Comment by Void2258** [comment]
Is the HRO working on the latent image or the rendered image? Those would produce different results, and working on the latent shouldn't be doable after an image has been rendered unless the latent were saved. There are a lot of settings that could be made available/explicit regarding the HRO process.

**Comment by aleccrichton** [comment]
Is this resolved? The HRO Option needs to have a 512px or relational width, its currently too small.

**Comment by psychedelicious** [comment]
HRO is not implemented in v3
